llvm/lib/TableGen/Error.cpp

```diff
@@ -170,4 +170,12 @@ void CheckAssert(SMLoc Loc, Init *Condition, Init *Message) {
   }
 }
 
+// Dump a message to stderr.
+void dumpMessage(SMLoc Loc, Init *Message) {
+  if (auto *MessageInit = dyn_cast<StringInit>(Message))
+    PrintNote(Loc, MessageInit->getValue());
+  else
+    PrintWarning(Loc, "dump message is not a string - use `!repr`");
```

Comment:
Given that it's the only failure mode with the only actionable fix, should we magically wrap non-string inputs in `!repr` and allow `dump` to print out anything?

Reply:
Done